R/user_error_checks.R

```diff
@@ -0,0 +1,45 @@
+# This function exists to check if a portfolio has a single holdings
+# date, or multiple. If there is a single date, then everything is good.
+# but on the platform, if a user groups multiple  portfolios, and some of
+# thse have different holding dates, then the parameters file will
+# contaain a yaml array with the distinct dates from each grouped
+# portfolio.
+# Examples:
+#
+# simple portfolio
+# holdings_date: 2020Q4
+#
+# groupedd portfolio with all same date:
+# holdings_date: 2020Q4
+#
+# grouped portfolio with different dates:
+# holdings_date: [2019Q4, 2020Q4]
+#
+# `config` will read this in as a character vector, so all we need to do
+# here is check for length > 1
+
+check_grouped_portfolio_years <- function(
+  port_holdings_date
+  ) {
```

Comment:
Not gonna get into style :-)

Reply:
Nit. This (and other functions here) seems to be called primarily for its side effects. Then it's nice to return the first argument invisibly so we can pipe it is we choose so.

https://design.tidyverse.org/out-invisible.html